docs/biovalidator_benchmarks/README.md

```diff
@@ -0,0 +1,87 @@
+new_objectset.add_new_json("../../examples/json_validation_tests/assay_valid-2_sequencing.json")
+# To count the number of items (i.e. objects) in the objectArray
+new_objectset.count_n_objects()
+# And finally to save the new object-set
+new_objectset.save_json(output_filepath = "my_new_object_set.json", add_suffix= True)
+````
+The idea of taking an existing object-set (e.g. ``object-set_valid-1.json``) as template is because it will have the correct structure of properties, and we will just have to empty it, populate it with the JSON objects we want, save it, and finally fill in the properties whose values we emptied.
+
+### ``create_benchmarks.py``
+The script can be called from the terminal:
+````bash
+$ python3 create_benchmarks.py --help
+usage: create_benchmarks.py [-h] --endpoint ENDPOINT --input_directory INPUT_DIRECTORY [--output_directory OUTPUT_DIRECTORY] [--number_of_iterations NUMBER_OF_ITERATIONS] [--n_parallel_threads N_PARALLEL_THREADS] [--stop_at_errors]
+                            [--not_just_summary_df] [--only_print_complete_df]
+
+Create benchmarks for a Biovalidator server. Check details at https://github.com/EbiEga/ega-metadata-schema/tree/main/docs/biovalidator_benchmarks.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  --endpoint ENDPOINT   The endpoint of the Biovalidator server. Example: "http://localhost:3020/validate"
+  --input_directory INPUT_DIRECTORY
+                        The filepath to a directory that contains the JSON documents to validate. Example: "examples/json_validation_tests/"
+  --output_directory OUTPUT_DIRECTORY
+                        The filepath to a directory in which the results will be stored. Example: "2023_benchmarks/". Default value: "./"
+  --number_of_iterations NUMBER_OF_ITERATIONS
+                        The amount of iterations of validation of each JSON document within the input directory. Example: 10. Default value: 1
+  --n_parallel_threads N_PARALLEL_THREADS
+                        The amount of parallel threads for each validation iteration. Example: 5. Default value: 1
+  --stop_at_errors      A boolean switch by which we tell the script to stop the execution when a validation error is raised.
+  --not_just_summary_df
+                        A boolean switch by which we tell the script to not just print the summary dataframe with parameters, but instead save the parameters in files and generate a README.
+  --only_print_complete_df
+                        A boolean switch by which we tell the script that we do not want any files to be saved and, instead, we just want the complete table of parameters to be printed.
+````
+There are three major ways in which this script is intended to be used:
+````bash
+# Just printing the summary of all validated files, to check that all are validated
+$ python3 create_benchmarks.py --endpoint "http://localhost:3020/validate" --input_directory "../../examples/json_validation_tests/"
```

Comment:
I'm afraid you won't be able to use the EGA service very often now: it's commonly down. And when you tried to use it, it was. And still is 😓 

Re. other warnings:

> The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.

I think refers to the following line at method ``iterate_over_dir()``. That warning never appeared with my version of pandas. 
````python3
self.complete_df = self.complete_df.append(self.iteration_list, ignore_index=True)
````
I have updated my pandas version and also adapted the method to use concat instead.

> ValueError: complete_df has not been initialized or is empty. Use method 'iterate_over_dir()' at least once before trying to create the summary dataframe.

This last one is an exception I placed there on purpose, which basically means that there is no data in the complete dataframe. There is none because the endpoint was not responding. If you see it valuable I could add a check on the validation to see if the endpoint responds correctly before doing anything else.

Reply:
For the last point, I think it would be helpful to somehow differentiate to the user when an error is because fetching from the endpoint didn't work, otherwise it looks like something is wrong with the code.

Reply:
Done 👍 
The script will stop and report that the endpoint is not reachable